// video-react/lib/components/PosterImage.js
amis.define('node_modules/video-react/lib/components/PosterImage', function(require, exports, module, define) {

  "use strict";
  
  var _interopRequireDefault = require("node_modules/@babel/runtime/helpers/interopRequireDefault");
  
  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports["default"] = void 0;
  
  var _propTypes = _interopRequireDefault(require("node_modules/prop-types/index"));
  
  var _react = _interopRequireDefault(require("node_modules/react/index"));
  
  var _classnames = _interopRequireDefault(require("node_modules/classnames/index"));
  
  var propTypes = {
    poster: _propTypes["default"].string,
    player: _propTypes["default"].object,
    actions: _propTypes["default"].object,
    className: _propTypes["default"].string
  };
  
  function PosterImage(_ref) {
    var poster = _ref.poster,
        player = _ref.player,
        actions = _ref.actions,
        className = _ref.className;
  
    if (!poster || player.hasStarted) {
      return null;
    }
  
    return _react["default"].createElement("div", {
      className: (0, _classnames["default"])('video-react-poster', className),
      style: {
        backgroundImage: "url(\"".concat(poster, "\")")
      },
      onClick: function onClick() {
        if (player.paused) {
          actions.play();
        }
      }
    });
  }
  
  PosterImage.propTypes = propTypes;
  PosterImage.displayName = 'PosterImage';
  var _default = PosterImage;
  exports["default"] = _default;

});
